api/src/main/java/ca/bc/gov/educ/studentdatacollection/api/calculator/impl/IndependentSchoolAndBandCodeCalculator.java

```diff
@@ -0,0 +1,50 @@
+package ca.bc.gov.educ.studentdatacollection.api.calculator.impl;
+
+import ca.bc.gov.educ.studentdatacollection.api.calculator.FteCalculator;
+import ca.bc.gov.educ.studentdatacollection.api.constants.v1.Constants;
+import ca.bc.gov.educ.studentdatacollection.api.constants.v1.SchoolCategoryCodes;
+import ca.bc.gov.educ.studentdatacollection.api.model.v1.BandCodeEntity;
+import ca.bc.gov.educ.studentdatacollection.api.repository.v1.BandCodeRepository;
+import ca.bc.gov.educ.studentdatacollection.api.struct.SdcStudentSagaData;
+import ca.bc.gov.educ.studentdatacollection.api.struct.v1.FteCalculationResult;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@Component
+@Slf4j
+public class IndependentSchoolAndBandCodeCalculator implements FteCalculator {
+    FteCalculator nextCalculator;
+    @Getter
+    private int processingSequenceNumber = 5;
+    @Autowired
+    BandCodeRepository bandCodeRepository;
+    @Override
+    public void setNext(FteCalculator nextCalculator) {
+        this.nextCalculator = nextCalculator;
+    }
+    @Override
+    public FteCalculationResult calculateFte(SdcStudentSagaData studentData) {
+        var isIndependentSchool = studentData.getSchool() != null && StringUtils.equals(studentData.getSchool().getSchoolCategoryCode(), SchoolCategoryCodes.INDEPEND.getCode());
+        Optional<BandCodeEntity> bandCode = StringUtils.isBlank(studentData.getSdcSchoolCollectionStudent().getBandCode()) ? (Optional.empty()): bandCodeRepository.findById(studentData.getSdcSchoolCollectionStudent().getBandCode());
```

Comment:
I think we don't need to check validity of band codes since that happens in validation rules. Infact validation rules will check validity of all code table values and flag it in case of mismatch

Reply:
Thanks, updated!